### PR TITLE
feat(auth): Adds support to specifying Auth state persistence.

### PR DESCRIPTION
**Pull request by bojeil-google**
Defines externs for new API to specify Auth state persistence.
fix(auth): Adds missing phoneNumber property in UserInfo.
fix(auth): Fixes incorrect return type in reauthenticateAndRetrieveDataWithCredential.

### CODE REVIEW

File: externs/firebase-auth-externs.js [hunk 1]
```diff
@@ -480,7 +487,7 @@ firebase.User.prototype.reauthenticateWithCredential = function(credential) {};
  * </dl>
  *
  * @param {!firebase.auth.AuthCredential} credential
- * @return {!firebase.Promise<void>}
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
```
Comment (jsayol) [inline externs/firebase-auth-externs.js]: This same change should be made in the typings.

File: externs/firebase-auth-externs.js [hunk 2]
```diff
@@ -788,6 +795,69 @@ firebase.auth.Auth.prototype.app;
 firebase.auth.Auth.prototype.currentUser;
 
 /**
+ * @enum {string}
+ * An enumeration of the possible persistence mechanism types.
+ */
+firebase.auth.Auth.Persistence = {
```
Comment (jsayol) [inline externs/firebase-auth-externs.js]: Same here, this should also be in the typings.

File: externs/firebase-auth-externs.js (hunk 1 above)
Comment (jshcrowthe) [inline externs/firebase-auth-externs.js]: Typings were autoimported from externs (same for below)

**Comment by jsayol** [comment]
Just FYI, commit e2d1a40 undid most of the typings improvements from PR #56

**Pull request merged**